EIPS/eip-5341.md

```diff
@@ -0,0 +1,117 @@
+
+### Simple payment to a single address
+This EIP does not specify the method of sending payment to referrers as there can be various ways of doing it. Attempting to do this as part of this standard would increase the implementation complexity and increase gas costs. This ERC should be considered a minimal, gas-efficient building block for further innovation in referral program.
+
+## Backwards Compatibility
+This standard is fully ERC-721 compatible by adding an extension function set.
+
+Besides, new functions introduced in this standard have many similarities with the existing functions in ERC-721, which helps developers to easily adopt the standard quickly.
+
+## Reference Implementation
+```solidity
+contract NFTReferralProgram is ERC721, INFTReferralProgram {
+    // The percentage of sale price referrer obtains
+    uint96 private _referralFraction;
+
+    // Mapping from tokenId to the referrer address
+    mapping(uint256 => address) internal _referrers;
+
+    /// @notice Set the referrer of a NFT
+    /// @param tokenId The tokend id of the NFT
+    /// @param referrer The referrer address of the token
+    function _setReferrer(uint256 tokenId, address referrer) internal virtual {
+        _referrers[tokenId] = referrer;
+    }
+
+    /// @notice Set the referral fraction of the NFT
+    /// @param referralFeeNumerator The referral fee as a percentage of sale price
+    function _setReferralFraction(uint96 referralFeeNumerator)
+        internal
+        virtual
+    {
+        require(
+            referralFeeNumerator <= _feeDenominator(),
+            "Referral fee will exceed salePrice"
+        );
+
+        _referralFraction = referralFeeNumerator;
+    }
+
+    /// @notice Get the referrer address of a tokenId
+    /// @param tokenId The tokend id of the NFT
+    function referrerOf(uint256 tokenId) public view virtual returns (address) {
+        return _referrers[tokenId];
+    }
+
+    /// @notice Get the referral fee for a sales price
+    /// @param salesPrice The sales price of a NFT
+    function referralFee(uint256 salesPrice)
+        public
+        view
+        virtual
+        override
+        returns (uint256)
+    {
+        return (salesPrice * _referralFraction) / _referralFeeDenominator();
+    }
+
+    /// @notice The denominator with which to interpret the fee set in {_setReferralFraction} as a fraction of the sale price.
+    /// Defaults to 10000 so fees are expressed in basis points, but may be customized by an override.
+    function _referralFeeDenominator() internal pure virtual returns (uint96) {
+        return 10000;
+    }
+```
+
+## Security Considerations
+The EIP sets the referral fraction and the token to referrer mapping. The owner can use them to transfer referral bonus to referrer when referee mints.
```

Comment:
So, you're saying that the project maintainers must be trusted to pay referrers out? If so, could you make that more explicit?